docs/sandboxing-via-appprojects.md

```diff
@@ -0,0 +1,224 @@
+# Sandboxing AppStudio user namespaces with AppProject
+
+This document examines the implementation details of how to add support for constraining AppStudio/Core GitOps Service users within an AppProject.
+
+Feature: [GITOPSRVCE-428](https://issues.redhat.com/browse/GITOPSRVCE-428)
+Epic: [GITOPSRVCE-56](https://issues.redhat.com/browse/GITOPSRVCE-56)
+
+
+# Introduction
+
+An **API namespace** is a Kubernetes Namespace where users of AppStudio can create AppStudio API CRs, and deploy their Applications.
+
+
+* These are the '*-tenant' Namespaces on the cluster.
+* For example, ‘jgwest-tenant' is mine. (Based on your Red Hat ID)
+
+Each API namespace has a corresponding **ClusterUser** row in the database.  ClusterUser is a simple mechanism for tracking individual users within the GitOps Service (this mechanism may evolve over time, for example, to support other ownership concepts,  but currently fits our needs in AppStudio).
+* There is a 1-1 relationship between a user’s API Namespace and ClusterUser row
+* For each user’s API namespace, there is a corresponding ClusterUser row that corresponds to that namespace, and vice versa.
+
+Thus, when thinking about how to constrain users, we can use the ClusterUser row (and relationships to that row) to represent all of the permissions that a user should have.
+
+The ClusterUser table is very simple:
```

Comment:
I think we need to mention new field which we have added recently to the ClusterUser table i.e, `display_name`  https://github.com/redhat-appstudio/managed-gitops/blob/main/db-schema.sql#L139 ?